library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
Yes, that is what I suggested.  Right now `p` would return ``[`n]`` (as a vm_obj) for a list of names, but we want to insert ``expr.app (expr.const `list.cons [..]) ..``.  Letting the parser extension do the construction of the pre-expression is just consistent with how we do it in the tactics.

We might want to provide a `mk_list_pexpr : list pexpr -> pexpr` function however.